### PR TITLE
chore: switch to mockery config file

### PR DESCRIPTION
**Pull request by ShreyaLnuHpe**
## Ticket

[RM-277](https://hpe-aiatscale.atlassian.net/browse/RM-277)

## Description

We are transitioning to using a single mockery configuration file to streamline our mock generation process. Currently, we have 18 individual mockery commands, which are time-consuming. By consolidating these into a single configuration file, we can significantly reduce execution time and improve efficiency.

## Test Plan

`make -C master clean`  to delete local mocks

`make -C master check` to generate mocks (also sometimes `make -C master mocks` )

`make -C master test-intg` to make sure mocks are generated right 
ran with `devcluster -c ./.circleci/devcluster/elastic-base.devcluster.yaml`
and confirmed if the devcluster ran fine - `curl -X GET "http://localhost:9200/_cluster/health?pretty"`

## Checklist

- [X] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[RM-277]: https://hpe-aiatscale.atlassian.net/browse/RM-277?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/Makefile [hunk 1]
```diff
@@ -88,25 +88,7 @@ stream-gen: $(STREAM_PYTHON_CLIENT) $(STREAM_TS_CLIENT)
...
-	mockery --quiet --name=JobInterface --srcpkg=k8s.io/client-go/kubernetes/typed/batch/v1 --output internal/mocks --filename job_iface.go
-	mockery --quiet --name=ResourceManager --dir=internal/rm --output internal/mocks --filename rm.go
-	mockery --quiet --name=AllocationService --dir=internal/task --output internal/mocks/allocationmocks --filename allocation_service.go --outpkg allocationmocks
-	mockery --quiet --name=ResourceManagerAuthZ --dir=internal/rm --output internal/mocks --filename rm_authz_iface.go
-	mockery --quiet --name=Interface --output internal/mocks --srcpkg "k8s.io/client-go/kubernetes" --filename k8s_clientset.go --structname K8sClientsetInterface
-	mockery --quiet --name=CoreV1Interface --output internal/mocks --srcpkg "k8s.io/client-go/kubernetes/typed/core/v1" --filename k8s_corev1_iface.go --structname K8sCoreV1Interface
-	mockery --quiet --name=BatchV1Interface --output internal/mocks --srcpkg "k8s.io/client-go/kubernetes/typed/batch/v1" --filename k8s_batchv1_iface.go --structname K8sBatchV1Interface
-
+	mockery --config "./.mockery.yaml"
```
Comment (NicholasBlaskey) [inline master/Makefile:91]: is the output a lot? 

Should we add ```--quite``` to this? I am fine either way

File: master/.mockery.yaml [hunk 2]
```diff
@@ -0,0 +1,124 @@
+# .mockery.yaml
+with-expecter: True
+dir: internal/mocks
+outpkg: mocks
+quiet: True
+packages:
+  github.com/determined-ai/determined/master/internal/db:
+    interfaces:
+      # mockery --quiet --name=DB --dir=internal/db --output internal/mocks --filename db.go
```
Comment (NicholasBlaskey) [inline master/.mockery.yaml:9]: I don't think we need these comments

**Review by NicholasBlaskey — APPROVED** [review]
LGTM

thanks for working on this

File: master/Makefile (hunk 1 above)
Comment (ShreyaLnuHpe) [reply master/Makefile:91]: We do not need it as it does not add any noise:

```
% make -C master check                        
rm -f build/schema_gen.stamp
rm -f ../harness/determined/common/streams/wire.py
rm -f ../webui/react/src/services/stream/wire.ts
../tools/scripts/gen-attributions.py master packaging/LICENSE
go generate ./pkg/schemas/...
mkdir -p build
touch build/schema_gen.stamp
go build -o build/stream-gen ./cmd/stream-gen
build/stream-gen ./internal/stream//authz_basic_impl.go ./internal/stream//subscription.go ./internal/stream//models.go ./internal/stream//test_util.go ./internal/stream//model_versions.go ./internal/stream//util.go ./internal/stream//supervisor.go ./internal/stream//socketlike.go ./internal/stream//messages.go ./internal/stream//projects.go ./internal/stream//authz_iface.go ./internal/stream//publisher.go --python --output ../harness/determined/common/streams/wire.py
build/stream-gen ./internal/stream//authz_basic_impl.go ./internal/stream//subscription.go ./internal/stream//models.go ./internal/stream//test_util.go ./internal/stream//model_versions.go ./internal/stream//util.go ./internal/stream//supervisor.go ./internal/stream//socketlike.go ./internal/stream//messages.go ./internal/stream//projects.go ./internal/stream//authz_iface.go ./internal/stream//publisher.go --ts --output ../webui/react/src/services/stream/wire.ts
mockery --config "./.mockery.yaml"
mkdir -p build
touch build/mock_gen.stamp
# Checking that committed, generated code is up-to-date by ensuring that
# git reports the files as unchanged after forcibly regenerating the files:
test -z "" || (git diff; false)
go mod tidy
git diff --quiet ../go.mod ../go.sum
golangci-lint --build-tags integration run --timeout 10m
WARN [linters_context] running gomodguard failed: unable to parse module file go.mod: go.mod:3: invalid go version '1.22.0': must match format 1.23: if you are not using go modules it is suggested to disable this linter 
```

**Pull request merged**